src/live_restore/live_restore_fs.c

```diff
@@ -794,20 +789,23 @@ __live_restore_fh_read(
          *
          *
          */
-        size_t dest_partial_read_len = (size_t)(hole_begin_off - offset);
-        size_t source_partial_read_len = len - dest_partial_read_len;
-
-        /* First read the serviceable portion from the destination. */
-        __wt_verbose_debug3(session, WT_VERB_LIVE_RESTORE, "%s", "    PARTIAL READ TRIGGERED");
-        WT_ERR(__live_restore_fh_read_destination(
-          session, lr_fh->destination, offset, dest_partial_read_len, read_data));
-
-        /* Now read the remaining data from the source. */
-        WT_ERR(__live_restore_fh_read_source(session, lr_fh->source, hole_begin_off,
-          source_partial_read_len, read_data + dest_partial_read_len));
-    } else
-        /* Read the full read from the source. */
-        WT_ERR(__live_restore_fh_read_source(session, lr_fh->source, offset, len, read_data));
+        if (hole_begin_off > offset) {
+            WT_ERR(__wt_calloc(session, 1, len, &tmp_buf));
+            size_t dest_partial_read_len = (size_t)(hole_begin_off - offset);
+            size_t source_partial_read_len = len - dest_partial_read_len;
+
+            /* First read the serviceable portion from the destination. */
+            WT_ERR(__live_restore_fh_read_destination(
+              session, lr_fh->destination, offset, dest_partial_read_len, tmp_buf));
+
+            /* Now read the remaining data from the source. */
+            WT_ERR(__live_restore_fh_read_source(session, lr_fh->source, hole_begin_off,
+              source_partial_read_len, tmp_buf + dest_partial_read_len));
+            WT_ASSERT_ALWAYS(session, WT_STRING_MATCH(read_data, tmp_buf, len),
```

Comment:
Minor, it's already under diagnostic, so n point of putting `WT_ASSERT_ALWAYS` here.